presto-orc/src/main/java/com/facebook/presto/orc/reader/BaseStreamReader.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.reader;
+
+import com.facebook.presto.orc.metadata.ColumnEncoding;
+import com.facebook.presto.orc.stream.InputStreamSources;
+
+import java.io.IOException;
+import java.util.List;
+
+public interface BaseStreamReader
```

Comment:
Having both `StreamReader` and `SelectiveStreamReader` inheriting `BaseStreamReader` is a bit confusing from the naming. Can we actually have:
- `BaseStreamReader ` -> `StreamReader`
- `StreamReader` -> `BatchStreamReader`
- `SelectiveStreamReader` -> `SelectiveStreamReader`

It would be easier to understand the behavior of the two stream readers. Also making all inheriting classes of `BatchStreamReader` having `Batch` in the name would be helpful to understand.